packages/video_player/video_player_avfoundation/example/ios/RunnerUITests/VideoPlayerUITests.m

```diff
@@ -60,4 +60,46 @@ - (void)testPlayVideo {
   }
 }
 
+- (void)testEncryptedVideoStream {
+  // This is to fix a bug (https://github.com/flutter/flutter/issues/111457) in iOS 16 with blank
+  // video for encrypted video streams.
+
+  NSString *tabName = @"Remote enc m3u8";
+
+  NSPredicate *predicate = [NSPredicate predicateWithFormat:@"label BEGINSWITH %@", tabName];
+  XCUIElement *unselectedTab = [self.app.staticTexts elementMatchingPredicate:predicate];
+  XCTAssertTrue([unselectedTab waitForExistenceWithTimeout:30.0]);
+  XCTAssertFalse(unselectedTab.isSelected);
+  [unselectedTab tap];
+
+  XCUIElement *selectedTab = [self.app.otherElements
+      elementMatchingPredicate:[NSPredicate predicateWithFormat:@"label BEGINSWITH %@", tabName]];
+  XCTAssertTrue([selectedTab waitForExistenceWithTimeout:30.0]);
+  XCTAssertTrue(selectedTab.isSelected);
+
+  // Wait until the video is loaded.
+  [NSThread sleepForTimeInterval:60];
+
+  NSMutableSet *frames = [NSMutableSet set];
+  int numberOfFrames = 60;
+  for (int i = 0; i < numberOfFrames; i++) {
+    XCUIScreenshot *screenshot = self.app.screenshot;
+
+    // Attach the screenshots for debugging if the test fails.
+    XCTAttachment *attachment = [XCTAttachment attachmentWithScreenshot:screenshot];
+    attachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+    [self addAttachment:attachment];
+
+    UIImage *image = screenshot.image;
+    [frames addObject:UIImagePNGRepresentation(image)];
+    // Take random interval between [1, 2) seconds, since the video length could be the same as a
+    // fixed interval, which would always result in the same frame.
+    NSTimeInterval sampleInterval = 1 + ((double)arc4random() / UINT32_MAX);
```

Comment:
I think this shouldn't be random just in case it flakes.  We have control of the example video so it shouldn't be between 1 and 2 seconds, and if it does change we want this test to consistently fail.